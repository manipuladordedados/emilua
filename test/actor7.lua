local println = require('println')
local sleep_for = require('sleep_for')

if _CONTEXT == 'main' then
    ch = spawn_vm('.')
    sleep_for(100)
    ch:send('foo')
    println('bar')
else
    assert(_CONTEXT == 'worker')
    local inbox = require('inbox')
    spawn(function()
        inbox:recv()
    end):detach()
    println(inbox:recv())
end
