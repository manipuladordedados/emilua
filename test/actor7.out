Fiber 0x1 panicked: 'Device or resource busy'
stack traceback:
	[string "?"]: in function 'recv'
	input.lua:13: in function <input.lua:12>
	[C]: in function ''
	[string "?"]: in function <[string "?"]:0>
foo
bar
